.devcontainer/devcontainer.json

```diff
@@ -1,33 +1,35 @@
 // For format details, see https://aka.ms/devcontainer.json. For config options, see the
-// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-docker-compose
 {
+	// Name for the dev container
+	"name": "langchain",
+
+	// Point to a Docker Compose file
 	"dockerComposeFile": "./docker-compose.yaml",
+
+	// Required when using Docker Compose. The name of the service to connect to once running
 	"service": "langchain",
+
+	// Sets the default path to open when connecting to the container
 	"workspaceFolder": "/workspaces/langchain",
-	"name": "langchain",
-	"customizations": {
-		"vscode": {
-			"extensions": [   
-				"ms-python.python"
-			],
-			"settings": {
-				"python.defaultInterpreterPath": "/home/vscode/langchain-py-env/bin/python3.11"
-			}
-		}
-		
-	},
-
-	// Features to add to the dev container. More info: https://containers.dev/features.
-	"features": {},
+
+	// Prevent the container from shutting down
+	"overrideCommand": true
+
+	// Features to add to the dev container. More info: https://containers.dev/features
+	// "features": {
+	// 	"ghcr.io/devcontainers-contrib/features/poetry:2": {}
+	// }
 
 	// Use 'forwardPorts' to make a list of ports inside the container available locally.
 	// "forwardPorts": [],
 
 	// Uncomment the next line to run commands after the container is created.
 	// "postCreateCommand": "cat /etc/os-release",
 
+	// Configure tool-specific properties.
+	// "customizations": {},
+
 	// Uncomment to connect as an existing user other than the container default. More info: https://aka.ms/dev-containers-non-root.
 	// "remoteUser": "devcontainer"
```

Comment:
Is the dev container config creating a `devcontainer` user? Not confident if uncommenting would work.

Reply:
Hmm I suppose it's creating a `vscode` user rather than one named `devcontainer`: https://github.com/devcontainers/images/blob/main/src/python/.devcontainer/devcontainer.json#L38.

What would be your recommendation?

Reply:
Maybe use the updated comment from the templates: https://github.com/devcontainers/templates/blob/main/src/python/.devcontainer/devcontainer.json

This changed when the base images started including metadata that sets the remoteUser.

Reply:
Ah good point, thanks! Updated to match the updated template's content.